src/Adapter/Order/CommandHandler/AddCartRuleToOrderHandler.php

```diff
@@ -49,82 +50,137 @@ final class AddCartRuleToOrderHandler extends AbstractOrderHandler implements Ad
     public function handle(AddCartRuleToOrderCommand $command): void
     {
         $order = $this->getOrderObject($command->getOrderId());
+        $orderInvoice = $this->createSingleOrderInvoice($order, $command);
 
-        // If the discount is for only one invoice
-        if ($order->hasInvoice() && null !== $command->getOrderInvoiceId()) {
-            $orderInvoice = new OrderInvoice($command->getOrderInvoiceId()->getValue());
-            if (!Validate::isLoadedObject($orderInvoice)) {
-                throw new OrderException('Can\'t load Order Invoice object');
+        $discountValue = (float) str_replace(',', '.', $command->getDiscountValue());
```

Comment:
😱

Reply:
Sorry guys, I have no knowledge about whats happening there deep, I just tried to at least separate to private methods, didn't wan't to break the logic.

Reply:
It is typed as null|Number . If I use Number I need to rework all calculations, thats risky and alot of code. You think its worth it ?